openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResolutionStatusResource.java

```diff
@@ -0,0 +1,292 @@
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "The test case failure status",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = TestCaseResolutionStatus.class)))
+      })
+  public ResultList<TestCaseResolutionStatus> listForSequenceId(
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Sequence ID", schema = @Schema(type = "UUID")) @PathParam("sequenceId")
+          UUID sequenceId) {
+    OperationContext operationContext = new OperationContext(Entity.TEST_CASE, MetadataOperation.VIEW_ALL);
+    ResourceContextInterface resourceContext = ReportDataContext.builder().build();
+    authorizer.authorize(securityContext, operationContext, resourceContext);
+
+    return repository.listTestCaseFailureStatusesForSequenceId(sequenceId);
+  }
+
+  @GET
+  @Path("/{id}")
+  @Operation(
+      operationId = "getTestCaseFailureStatusById",
+      summary = "Get test case failure status by id",
+      description = "Get a test case failure status by id",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "The test case failure status",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = TestCaseResolutionStatus.class)))
+      })
+  public TestCaseResolutionStatus get(
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Test Case Failure Status ID", schema = @Schema(type = "UUID")) @PathParam("id")
+          UUID testCaseFailureStatusId) {
+    OperationContext operationContext = new OperationContext(Entity.TEST_CASE, MetadataOperation.VIEW_ALL);
+    ResourceContextInterface resourceContext = ReportDataContext.builder().build();
+    authorizer.authorize(securityContext, operationContext, resourceContext);
+
+    return repository.getById(testCaseFailureStatusId);
+  }
+
+  @POST
+  @Operation(
+      operationId = "createTestCaseFailureStatus",
+      summary = "Create a new test case failure status",
+      description = "Create a new test case failure status",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "The created test case failure status",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = CreateTestCaseResolutionStatus.class)))
+      })
+  public Response create(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Valid CreateTestCaseResolutionStatus createTestCaseFailureStatus) {
+    OperationContext operationContext = new OperationContext(Entity.TEST_CASE, MetadataOperation.EDIT_TESTS);
```

Comment:
We should also check the ownership of the test case.
Also what happens if another user wants to open a ticket on a test case failure. 
They may not have test case edit option